src/test/java/com/hillel/items_exchange/service/UserServiceTest.java

```diff
@@ -0,0 +1,166 @@
+        //given
+        UserChangePasswordDto userChangePasswordDto =
+                createUserChangePasswordDtoWithWrongOldPassword();
+        //then
+        InvalidDtoException invalidDtoException = assertThrows(InvalidDtoException.class,
+                () -> userService.updateUserPassword(userChangePasswordDto, userWithOldPassword));
+        assertTrue(invalidDtoException.getMessage().contains(
+                "Incorrect password. Please, check if you entered password correctly"));
+        verify(userRepository, never()).saveAndFlush(any());
+    }
+
+    @Test
+    void testUpdateUserPassword_WhenPasswordConfirmationWrong_ShouldThrowIllegalArgumentException() {
+        //given
+        UserChangePasswordDto userChangePasswordDto =
+                createUserChangePasswordDtoWithWrongPasswordConfirmation();
+        //then
+        assertThrows(IllegalArgumentException.class,
+                () -> userService.updateUserPassword(userChangePasswordDto, userWithOldPassword));
+    }
+
+    @Test
+    void testUpdateUserEmail_Successfully() {
+        //given
+        UserChangeEmailDto userChangeEmailDto =
+                createUserChangeEmailDtoWithCorrectData();
+        User userWithNewEmail = createUserWithNewEmail();
+        //when
+        when(userRepository.saveAndFlush(userWithOldPassword)).thenReturn(userWithNewEmail);
+        UserDto userDto = userService.updateUserEmail(userChangeEmailDto, userWithOldPassword);
+        //then
+        assertEquals(userDto.getUsername(), userWithNewEmail.getUsername());
+        verify(userRepository).saveAndFlush(userWithOldPassword);
+    }
+
+    @Test
+    void testUpdateUserEmail_WhenEmailConfirmationWrong_ShouldThrowIllegalArgumentException() {
+        //given
+        UserChangeEmailDto userChangeEmailDto =
+                createUserChangeEmailDtoWithWrongEmailConfirmation();
+        //then
+        assertThrows(IllegalArgumentException.class,
+                () -> userService.updateUserEmail(userChangeEmailDto, userWithOldPassword));
+    }
+
+    private User createUserWithOldPassword() {
+        userWithOldPassword = new User();
+        userWithOldPassword.setPassword(bCryptPasswordEncoder.encode(CORRECT_OLD_PASSWORD));
+
+        return userWithOldPassword;
+    }
+
+    private User createUserWithNewPassword() {
+        User user = new User();
+        user.setPassword(NEW_PASSWORD);
+
+        return user;
+    }
+
+    private UserChangePasswordDto createUserChangePasswordDtoWithCorrectData() {
+
+        return createUserChangePasswordDto(CORRECT_OLD_PASSWORD,
+                NEW_PASSWORD,
+                NEW_PASSWORD);
+    }
+
+    private UserChangePasswordDto createUserChangePasswordDtoWithWrongOldPassword() {
+
+        return createUserChangePasswordDto(WRONG_OLD_PASSWORD,
+                NEW_PASSWORD,
+                NEW_PASSWORD);
+    }
+
+    private UserChangePasswordDto createUserChangePasswordDtoWithWrongPasswordConfirmation() {
```

Comment:
The main idea of 3 methods that you've provided here: **createUserChangePasswordDtoWithCorrectData**, **createUserChangePasswordDtoWithWrongOldPassword**, **createUserChangePasswordDtoWithWrongPasswordConfirmation**
is to invoke the same method but with different parameters. You should remove them all and invoke **createUserChangePasswordDto** method with needed parameters directly from the tests.